### DISCUSSION TITLE
configuration of image size

### DISCUSSION THREAD

**Discussion by miullu** [SmilyOrg/photofield#127] [Q&A] [answered]
I don't understand how to configure the preview image, is it part of the thumbnail or required configuring the whole part of media source?

the preview image size can be configured in url by "?image_height=170",
is there more formal and convenient way?

Also, how do I set the resolution of the preview image?

**Comment by SmilyOrg** [comment]
Hey, I'm not sure what you mean with "preview image"?

Normally thumbnails are created while indexing if none exist already, then while rendering, the closest thumbnail size is picked and re-encoded to fit the screen size automatically.

**Reply by miullu** [reply]
How does it choose the thumbnail size? Due to the quality of the display on the web, I think I have some configuration errors

**Reply by SmilyOrg** [reply, marked as answer]
It uses [a few different heuristics](https://github.com/SmilyOrg/photofield/blob/main/io/io.go#L156-L193) to choose which thumbnail to use amongst the available sources. Some of this is configurable and some of it isn't. The thumbnail sources are configured under [sources in the config](https://github.com/SmilyOrg/photofield/blob/main/defaults.yaml#L238-L342).

I'm assuming you don't have any other pregenerated thumbnails (like from Synology Moments), in which case you'll most likely be left with three sources:

1. `sqlite`: pregenerated thumbnails in `photofield.thumbs.db` - [hard-coded to 256px](https://github.com/SmilyOrg/photofield/blob/main/io/sqlite/sqlite.go#L65)
2. `goexif`: embedded thumbnails in the jpegs - depends on camera, but usually 100-300px
3. `image` / `ffmpeg` : original photos resized on the fly - whatever is needed

While you can't configure the thumbnail size itself, you can influence which thumbnail is chosen for rendering. As an extreme example, if you set [sqlite thumbnail loading time](https://github.com/SmilyOrg/photofield/blob/main/defaults.yaml#L212) to something like `10000ms` instead of `6ms`, it will prefer loading all the other sources, unless they are estimated to take more than 10 seconds to load.

If you can paste a screenshot and/or your current configuration I can see if there is something that looks out of place (you can censor any private details). Generally it should look like the [demo](https://demo.photofield.dev/) by default, but I'm not sure if you consider that one good or bad quality :)
